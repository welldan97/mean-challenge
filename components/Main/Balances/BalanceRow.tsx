import { memo } from 'react';
import config from '../../../config';
import { Balance } from './types';

interface BalanceRowProps {
  balance: Balance;
}

const BalanceRow = memo(({ balance }: BalanceRowProps) => (
  <tr>
    <td>
      {new Intl.DateTimeFormat('pt-PT', {
        dateStyle: 'short',
        timeStyle: 'short',
      }).format(balance.dateTime.toJSDate())}
    </td>
    <td>{balance.averageBalance / 1000000000}</td>
    {config.showExtraData && <td>{balance.transactionsCount}</td>}
  </tr>
));

export default BalanceRow;
