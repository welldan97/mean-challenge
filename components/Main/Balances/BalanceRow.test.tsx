import { render } from '@testing-library/react';

import { balance } from '../../../lib/test/fixtures/balances';
import BalanceRow from './BalanceRow';

describe('BalanceRow', () => {
  test('render', () => {
    const { container } = render(
      <table>
        <tbody>
          <BalanceRow balance={balance} />
        </tbody>
      </table>,
    );

    expect(container).toMatchSnapshot();
  });
});
