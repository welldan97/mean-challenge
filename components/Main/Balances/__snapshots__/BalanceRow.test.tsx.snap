// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BalanceRow render 1`] = `
<div>
  <table>
    <tbody>
      <tr>
        <td>
          05/10/22, 20:00
        </td>
        <td>
          3.54946716
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
